iceoryx2-bb/posix/src/directory.rs

```diff
@@ -321,6 +321,31 @@ impl Directory {
         &self.path
     }
 
+    /// Removes an empty directory. If the directory is not empty it returns an error.
+    pub fn remove_empty(path: &Path) -> Result<(), DirectoryRemoveError> {
+        let msg = format!("Unable to remove empty directory \"{}\"", path);
+
+        if unsafe { posix::rmdir(path.as_c_str()) } == -1 {
+            handle_errno!(DirectoryRemoveError, from "Directory::remove",
```

Comment:
```suggestion
        if unsafe { posix::rmdir(path.as_c_str()) } == -1 {
            let msg = format!("Unable to remove empty directory \"{}\"", path);

            handle_errno!(DirectoryRemoveError, from "Directory::remove",
```
No need to create the string in the good path